ts/main.ts

```diff
@@ -1 +1,31 @@
-/* global data */
+document.addEventListener('DOMContentLoaded', () => {
+  const $form = document.getElementById('entry-form') as HTMLFormElement;
+  const $titleInput = document.getElementById('title') as HTMLInputElement;
+
+  const $photoUrlInput = document.getElementById(
+    'photo-url',
+  ) as HTMLInputElement;
+  const $notesInput = document.getElementById('notes') as HTMLTextAreaElement;
+  const $photoPreview = document.getElementById(
+    'photo-preview',
+  ) as HTMLImageElement;
+
+  $form.addEventListener('submit', (event: Event) => {
```

Comment:
Your image isn't updating when you paste the url in the input box. You should have an event listener that updates the image based on the change.